<!-- Assumes Gazebo is already running -->
<launch>
  <!-- Settings (Robot position args are overidden by spawn_world.launch)-->
  <arg name="x" default="0.0" />
  <arg name="y" default="0.0" />
  <arg name="z" default="0.0" />
  <arg name="yaw" default="0.0" />
  <!-- Rotation in radians. -->
  <arg name="level" default="level1" />
  <arg name="navigate" default="true" />

  <arg name="model_name" default="pioneer" />

  <!-- Launch respective robot -->
  <include file="$(find uml_3d_race)/launch/robots/spawn_$(arg model_name).launch">
    <arg name="x" value="$(arg x)" />
    <arg name="y" value="$(arg y)" />
    <arg name="z" value="$(arg z)" />
    <arg name="yaw" value="$(arg yaw)" />
    <!-- Rotation in radians. -->
    <arg name="level" value="$(arg level)" />
    <arg name="navigate" value="$(arg navigate)" />
  </include>

  <!-- Publish the two different goals needed for the given .world -->
  <node name="goal_publisher" pkg="uml_3d_race" type="goal_publisher" output="screen" ns="$(arg model_name)">
    <!-- Goal Positions -->
    <remap from="~goal_x" to="/goal_x" />
    <remap from="~goal_y" to="/goal_y" />
    <remap from="~spawn_x" to="/spawn_x" />
    <remap from="~spawn_y" to="/spawn_y" />
  </node>
  
</launch>
